docker/controller/Dockerfile

```diff
@@ -1,8 +1,32 @@
-FROM centos
+# Start from the latest golang base image
+FROM golang:latest as builder
 
-RUN mkdir -p /opt/tyk-k8s
-COPY files/build/tyk-k8s /opt/tyk-k8s/tyk-k8s
+# Set the Current Working Directory inside the container
+WORKDIR /app
+
+# Copy go mod and sum files
+COPY go.mod go.sum ./
+
+# Download all dependencies. Dependencies will be cached if the go.mod and go.sum files are not changed
+RUN go mod download
+
+# Copy the source from the current directory to the Working Directory inside the container
+COPY . .
+
+# Build the Go app
+RUN CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o tyk-k8s .
 
+
+######## Start a new stage from scratch #######
+FROM alpine:latest
```

Comment:
Please pin the version.

Also I'm a little wary about Alpine as a main image as it brings some non-standard system components (notably musl instead of glibc) that can affect functionality. For example, there was a known DNS trouble, seems to be resolved now in k8s, but indicates.

Debian slim might be a safer choice for the default image, while only slightly larger:
https://hub.docker.com/layers/debian/library/debian/buster-slim/images/sha256-7c459309b9a5ec1683ef3b137f39ce5888f5ad0384e488ad73c94e0243bc77d4?context=explore